### ISSUE TITLE
Demo Project: Add a test to verify that recharts can be tested with jest 

### ISSUE THREAD

**Issue by nikolasrieble** [recharts/recharts#3897]
Following up https://github.com/recharts/recharts/issues/3704

While we start using vitest, our users might still use jest. Lets verify that jest runs fine in the demo project.

**Comment by akamfoad** [comment]
I wouldn't bet on that to catch any bugs, would've been more assuring if the example were an independent app/repo, but the demo project currently embedded in our source code like another folder in the project.

What do you think?

**Comment by ckifer** [comment]
Yeah agreed, it needs to be a totally separate repo that installs recharts from npm

**Comment by nikolasrieble** [comment]
Why exactly should it be a separate repository? We could setup our demo project as a completely separate project in a mono repo. Having multiple repos seems to be a safe way for it being ignored / hardly ever used.

**Comment by ckifer** [comment]
Ah I guess when I say separate repo I mean installs from npm like a user would and doesn't import anything from our src directly

**Comment by nikolasrieble** [comment]
> Ah I guess when I say separate repo I mean installs from npm like a user would and doesn't import anything from our src directly

If we had the project in the same repository, we could have a workflow with yalc to publish from one project to another, and then run the test 🤔 Would that not be the perfect setup for our goal?

**Comment by akamfoad** [comment]
Even though I like monorepo setup, I think this would put a lot inside the recharts project.

We could have the demo in a separate repo, have a workflow with a cron job that runs every so often and runs its tests. It is possible to make this workflow install from npm nightly or canary tag.

This will need recharts to have an npm release often to the mentioned npm.

Having to download from npm better augments a normal user's flow of downloading and running recharts in such an environment.

**Comment by ckifer** [comment]
I kind of agree with @akamfoad here. Adding another project to recharts itself feels like bloat. This test project doesn't need to change as often as say, the docs do, and by much fewer people than touch the docs. 

It doesn't introduce the same issues that the docs site does. This, at most, needs changed when we add a new chart type. Updates can be automated, we can import * by default to check for typescript issues at build time. It won't save us from everything and shouldn't be meant to. For example we could have a similar test repo for next - which would do the same thing with different tooling

**Comment by branberry** [comment]
On second thought, I am not as opposed to including the demo apps within the Recharts repository. This looks like something [Vitest does](https://github.com/vitest-dev/vitest/tree/4c957a27234cddda272d66c37300b8fa90370b6f/examples) as well.

 One benefit is that we wouldn't need to wait to cut a release to see if the demo project still works as expected, and it could be a part of the CI/CD process. I think it would be nice to catch issues earlier as opposed to waiting to release a nightly build. I guess, how would we like to maintain the nightly branch? Would it be separate from master?

**Comment by ckifer** [comment]
If we use the same repo without installing from npm we lose all the benefit of doing this (imo). Last time consumers broke with next/jest issues we had no idea until bug reports started rolling in. 

But the local publish and install is what will allow us to do this on every commit or at least daily as long as that's possible in CI

**Comment by akamfoad** [comment]
What I had in mind initially was that
1. Recharts will publish a `canary` release every time a PR merged on master
2. The demo project (separate repo) will run the tests on that latest on canary from recharts
   - This can be done by utilizing a CRON job to run the tests, we can optimize this so it'll not run too soon or too late between releases
   - This can also be done by repository dispatch, this will define a workflow that can be dispatched by a workflow from another repo (recharts -> demo)

This will have the benefit of being something similar to a normal user's flow.

**Comment by ckifer** [comment]
Yeah that works too - canary releases would be cool to give some brave consumers another place to test out changes

**Comment by branberry** [comment]
Okay, gotcha! I think I might just have a misunderstanding. What do we lose by having this be a part of the repository as opposed to its own stand alone repository? Apologies for the confusion.

**Comment by ckifer** [comment]
An exact replication of what the consumer goes through. When we're in the recharts repo we have specific babel, typescript, test framework settings that make things work. Imports get updated when we change them, tests run on the files before the "build". Users run tests after the "build" from different entry points (lib, or es6 folders mostly) we can't regression test how these things work together without doing a release or mock release first

**Comment by branberry** [comment]
Okay, gotcha! Now I understand. Thank you for the clarification!

**Comment by PavelVanecek** [comment]
Now available here: https://github.com/recharts/recharts-integ/tree/main/integrations/ts4-react17
